packages/replay/src/replay.ts

```diff
@@ -426,9 +438,12 @@ export class ReplayContainer implements ReplayContainerInterface {
 
     const activityTime = Date.now();
 
-    // eslint-disable-next-line no-console
-    const log = this.getOptions()._experiments.traceInternals ? console.info : logger.info;
-    __DEBUG_BUILD__ && log(`[Replay] Converting buffer to session, starting at ${activityTime}`);
+    logInfo('[Replay] Converting buffer to session', this._options._experiments.traceInternals);
+
+    // Change to 'session' recording mode
+    // We do this before `flushImmediate` so that if this is called again while flushing is in progress,
+    // we do not repeat these steps
+    this.recordingMode = 'session';
```

Comment:
I moved this up here, because I think it _may_ lead to potential issues that this is only set after we await the `flushImmediate`.